packages/firestore/test/integration/api/database.test.ts

```diff
@@ -1098,6 +1101,21 @@ apiDescribe('Database', (persistence: boolean) => {
     });
   });
 
+  it('can unlisten queries after shutdown', async () => {
```

Comment:
Please move these changes to a different PR. We don't want to revert two PRs worth of work if we happen to find a problem with one of them.